doc/gui/extension/example_library/front-end/src/BasicTable.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useMemo, useState } from "react";
+import {
+    createRequestDataUpdateAction,
+    RowType,
+    RowValue,
+    useDispatch,
+    useDispatchRequestUpdateOnFirstRender,
+    useModule,
+} from "taipy-gui";
+
+interface BasicTableProps {
+    id?: string;
+    updateVarName?: string;
+    updateVars?: string;
+    data: Record<string, Record<string, unknown>>;
+    rowsPerPage?: number;
+}
+
+const BasicTable = (props: BasicTableProps) => {
+    const {
+        data,
+        rowsPerPage,
+        updateVarName = "",
+        updateVars = "",
+        id
+    } = props;
+    const [value, setValue] = useState<Record<string, unknown>>({});
+    const [currentPage, setCurrentPage] = useState(1);
+    const dispatch = useDispatch();
+    const module = useModule();
+    const refresh = data?.__taipy_refresh !== undefined;
+    useDispatchRequestUpdateOnFirstRender(dispatch, id, module, updateVars, updateVarName);
+
+    // Memoize column order
+    const [colsOrder] = useMemo(() => {
+        const colsOrder = Object.keys(value || {});
+        return [colsOrder, value || {}];
+    }, [value]);
+
+    // Memoize rows
+    const rows = useMemo(() => {
+        const rows: RowType[] = [];
+        if (value) {
+            colsOrder.forEach(
+                (col) =>
+                    value[col] &&
+                    (value[col] as RowValue[]).forEach(
```

Comment:
the type would be inferred if value is correctly typed

Reply:
Either I have to infer it as RowValue[] here or I have to infer the type for the value in forEach as RowValue, somehow it didn't recognize it at first.